hera_cal/io.py

```diff
@@ -2009,9 +2045,14 @@ def write_vis(fname, data, lst_array, freq_array, antpos, time_array=None, flags
 
     time_array : type=ndarray, contains unique Julian Date time bins of data (center of integration).
 
-    flags : type=DataContainer, holds data flags, matching data in shape.
+    flags : type=DataContainer or array, holds data flags, matching data in shape. If
+            an array, must be shape (Nbls, Ntimes, Nfreqs, Npols).
 
-    nsamples : type=DataContainer, holds number of points averaged into each bin in data (if applicable).
+    nsamples : type=DataContainer or array, holds number of points averaged into each bin in data 
+               (if applicable). If an array, must be shape (Nbls, Ntimes, Nfreqs, Npols).
+
+    data_array : type=array, if given, use this data as the data_array, instead of 
```

Comment:
This feels very hacky. Is it necessary? And if so, why aren't we also doing it with nsamples and flags?

An alternative is a bool (default False) where the user can put in arrays for `data`, `flags`, and `nsamples` and they are treated as such.

Reply:
I no longer use this in the LST-binner so I've removed these changes. It might be useful one day in the future, but we should be more careful than I was if we want to add it then. I did leave in the change that updates from using a map of a lambda to just doing a for loop.